src/schemes/boundary/open_boundary/system.jl

```diff
@@ -548,34 +548,23 @@ end
 @inline function transform_particle!(fluid_system::FluidSystem, system,
                                      boundary_zone, particle, v, u, v_fluid, u_fluid)
     # Activate particle in boundary zone
-    activate_particle!(system, fluid_system, particle, v, u, v_fluid, u_fluid)
+    activate_particle!(system, particle, v, u, v_fluid, u_fluid)
 
     # Deactivate particle in interior domain
     deactivate_particle!(fluid_system, particle, u_fluid)
 
     return fluid_system
 end
 
-@inline function activate_particle!(system_new, system_old, particle_old,
-                                    v_new, u_new, v_old, u_old)
-    particle_new = available_particle(system_new)
+@inline function activate_particle!(system_new, particle_old, v_new, u_new, v_old, u_old)
+    particle_new = activate_next_particle(system_new)
 
-    # Exchange densities
-    density = particle_density(v_old, system_old, particle_old)
-    set_particle_density(particle_new, v_new, system_new, density)
-
-    # Exchange pressure
-    pressure = particle_pressure(v_old, system_old, particle_old)
-    set_particle_pressure(particle_new, v_new, system_new, pressure)
-
-    # Exchange position and velocity
-    for dim in 1:ndims(system_new)
-        u_new[dim, particle_new] = u_old[dim, particle_old]
-        v_new[dim, particle_new] = v_old[dim, particle_old]
+    # Exchange integrated quantities
+    for i in axes(u_new, 1)
+        u_new[i, particle_new] = u_old[i, particle_old]
+        v_new[i, particle_new] = v_old[i, particle_old]
```

Comment:
You need two loops. One over `axes(u_new, 1)` for `u` and one over `axes(v_new, 1)` for `v` to include density/pressure.

Reply:
Oh right thanks!

Reply:
Ah now I remember why I didn't do it that way. 

`v_new` doesn't have the same size as `v_old` because density and pressure of the open boundary particles is extrapolated from the fluid domain and not evolved over time...